Documentation/RepoCompose.md

```diff
@@ -0,0 +1,289 @@
+                    }
+                },
+                "file": {
+                    "nuget.exe":  {
+                        "kind": "url",
+                        "uri": "https://dist.nuget.org/win-x86-commandline/latest/nuget.exe"
+                    }
+                }
+            },
+            "toolset": { 
+            },
+            "static": { 
+            }
+        },
+        "prereq": { 
+            "Microsoft Visual Studio": "2015",
+            "CMake" : "1.0"
+        } 
+    }
+}
+```
+
+### produces
+
+``` json
+{
+    "os-all": {
+        "nuget": {
+            "packages" {
+                "MicroBuild.Core": "0.2.0",
+                "Microsoft.NETCore.Platforms": "1.0.1"
+            }
+        },
+        "file": {
+            "nuget.exe": { } 
+        }
+    }
+}
+```
+
+## Implementation Stages
+
+Fully implementing the commands described in this document requires a decent amount of work.  The expectation is that it will be implemented in stages.  Each stage provides benefit to an existing tool or allows us to write a new tool for a major scenario:
+
+- Stage 1: NuGet build information.  NuGet packages are how the majority of repo artifacts are shared today.  Getting the NuGet section of the above commands implemented allows us to automate the majority of our composed builds.  This is really just a formalization of the commands repos already implement as a part of being inside the Maestro tool.  
+- Stage 2: Remainder of artifacts:  Fill out the build and toolset sections with all of the other file / NuGet dependencies.  This output will allow us to fully understand the tools required for composed builds.  
+- Stage 3: Pre-req information.  Having correct prereq information will allow us to understand how to create and provision build machines.
+- Stage 4: Change command for all artifacts.  The ability to change a dependency from downloading nuget.exe from nuget.org to a place on the local file system.  This will give us the ability to implement fully offline builds.  
+
+## FAQ
+
+### How would a composer tool match the inputs / outputs of repos?
+
+The goal of this effort is to allow a composer tool to examine an arbitrary set of repos and establish a build order.  It can do so by examining the output of `consumes` and `produces` for each repo and establishing a dependency graph based on the outputs.  Artifacts from `produces` can be linked to floating build dependencies in `consumes`.
+
+But there is no guarantee that at any given point in time that the outputs of one repo will match exactly with the inputs of a depndent repo.  The names are likely to match but not the versions.  For example what is the chance for any given commit that dotnet/corefx is outputing System.Collections.Immutable at the exact version that dotnet/roslyn is cosuming for a given commit?  Probably fairly unlikely and getting them to agree requires coordination which is expensive and intended to be avoided by this very design.
```

Comment:
Doh!  If it's not abundantly clear yet, grammar is not my strong suit :frowning: 

I'm going to submit a follow up PR later today to fix all the issues you found.